lib/Alchemy/Phrasea/SearchEngine/Elastic/Search/QueryVisitor.php

```diff
@@ -299,43 +293,33 @@ private function visitBoolean(TreeNode $node)
         }
     }
 
-    private function visitDatabaseNode(Element $element)
-    {
-        if ($element->getChildrenNumber() !== 1) {
-            throw new \Exception('Base filter can only have a single child.');
-        }
-        $baseName = $element->getChild(0)->getValue()['value'];
-
-        return new AST\DatabaseExpression($baseName);
-    }
-
-    private function visitCollectionNode(Element $element)
+    private function visitNativeKeyValueNode(TreeNode $node)
     {
-        if ($element->getChildrenNumber() !== 1) {
-            throw new \Exception('Collection filter can only have a single child.');
+        if ($node->getChildrenNumber() !== 2) {
+            throw new \Exception('Key value expression can only have 2 childs.');
         }
-        $collectionName = $element->getChild(0)->getValue()['value'];
-
-        return new AST\CollectionExpression($collectionName);
+        $key = $this->visit($node->getChild(0));
+        $value = $this->visit($node->getChild(1));
+        return new AST\KeyValue\Expression($key, $value);
     }
 
-    private function visitTypeNode(Element $element)
+    private function visitNativeKeyNode(Element $element)
     {
         if ($element->getChildrenNumber() !== 1) {
-            throw new \Exception('Type filter can only have a single child.');
```

Comment:
What about changing `Exception` to `Alchemy\Phrasea\SearchEngine\Query\Exception\LogicException`?
Not very important but would be easier to detect whether it is an error in query logic compared to a code error